src/Adapter/Product/ProductUpdater.php

```diff
@@ -91,15 +92,58 @@ public function refreshProductCustomizabilityProperties(Product $product): void
         );
     }
 
+    /**
+     * @param Product $product
+     * @param int $defaultSupplierId
+     */
+    public function updateProductDefaultSupplier(Product $product, int $defaultSupplierId): void
+    {
+        if ($product->hasCombinations() || !$defaultSupplierId) {
+            $fieldsToUpdate['supplier_reference'] = '';
+            $fieldsToUpdate['wholesale_price'] = 0;
+        } elseif ($defaultSupplierId && !$product->hasCombinations()) {
+            $fieldsToUpdate['supplier_reference'] = ProductSupplier::getProductSupplierReference(
```

Comment:
Should we check that the default supplier exists? And we could also check here that the default supplier is one the suppliers associated to this product?